.github/workflows/ci.yml

```diff
@@ -69,39 +46,27 @@ jobs:
           sudo apt-get install -y libdrm-dev
       - name: Fetch drm headers
         run: ./.github/workflows/fetch_headers.sh
-      - name: Update deps
-        uses: actions-rs/cargo@v1
-        with:
-          command: update
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@1.70.0
         with:
-          toolchain: 1.66.0
-          profile: minimal
           components: clippy
-          default: true
-          override: true
-      - name: Downgrade (transitive) dependencies
-        run: |
-          cargo update -p home --precise 0.5.5
-          cargo update -p fdeflate --precise 0.3.5
-          cargo update -p flate2 --precise 1.0.35
+        # TODO: Remove once our MSRV is past 1.84 for the new resolver
+      - uses: dtolnay/rust-toolchain@1.84.0
+      - name: Downgrade msrv aware
+        run: env CARGO_RESOLVER_INCOMPATIBLE_RUST_VERSIONS=fallback cargo +1.84 generate-lockfile
       - name: Cargo cache
         uses: actions/cache@v3
         with:
           path: |
             ~/.cargo/registry
             ~/.cargo/git
-          key: ${{ runner.os }}-cargo-rust_1.66.0-${{ hashFiles('**/Cargo.toml') }}
+          key: ${{ runner.os }}-cargo-rust_1.70.0-${{ hashFiles('**/Cargo.toml') }}
       - name: Build cache
         uses: actions/cache@v3
         with:
           path: target
-          key: ${{ runner.os }}-build-rust_1.66.0-check-${{ hashFiles('**/Cargo.toml') }}
+          key: ${{ runner.os }}-build-rust_1.70.0-check-${{ hashFiles('**/Cargo.toml') }}
       - name: Clippy
-        uses: actions-rs/clippy-check@v1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: --all --all-features --all-targets -- -D warnings -A clippy::redundant_static_lifetimes
+        run: cargo +1.70.0 clippy --all --all-features --all-targets -- -D warnings -A clippy::redundant_static_lifetimes
```

Comment:
I'm only noticing now that we were only running `clippy` on our MSRV and not on `stable`. Is that intended?

Perhaps reduces contributor churn by not flagging small new contributions as failing CI (on irrelevant existing hunks in this codebase) if the repo hasn't been touched for a while and there's a new `stable` with stricter lints (similar to the stuff we had to do in #225 to get it back to a functioning state).